src/views/templates/utils/selectors.ts

```diff
@@ -1,29 +1,34 @@
 import { V1Template } from '@kubevirt-utils/models';
 import { getAnnotation } from '@kubevirt-utils/resources/shared';
 import { getLabel } from '@kubevirt-utils/resources/shared';
-import { WORKLOADS, WORKLOADS_LABELS } from '@kubevirt-utils/resources/template';
+import {
+  getTemplateVirtualMachineObject,
+  WORKLOADS_LABELS,
+} from '@kubevirt-utils/resources/template';
 import { VM_WORKLOAD_ANNOTATION } from '@kubevirt-utils/resources/vm/utils';
 
 import { ANNOTATIONS, LABELS } from './constants';
 
 export const getAffinity = (template: V1Template) =>
-  template?.objects[0]?.spec?.template?.spec?.affinity || {};
+  getTemplateVirtualMachineObject(template)?.spec?.template?.spec?.affinity || {};
 
 export const getEvictionStrategy = (template: V1Template): string =>
-  template?.objects[0]?.spec?.template?.spec?.evictionStrategy;
+  getTemplateVirtualMachineObject(template)?.spec?.template?.spec?.evictionStrategy;
 
 export const getNodeSelector = (template: V1Template) =>
-  template?.objects[0]?.spec?.template?.spec?.nodeSelector;
+  getTemplateVirtualMachineObject(template)?.spec?.template?.spec?.nodeSelector;
 
 export const getTemplateProviderName = (template: V1Template): string =>
   getAnnotation(template, ANNOTATIONS.providerName, null) ||
   getAnnotation(template, ANNOTATIONS.providerDisplayName, null);
 
-export const getTemplateWorkload = (template: V1Template): WORKLOADS =>
-  template?.objects[0]?.spec?.template?.metadata?.annotations?.[VM_WORKLOAD_ANNOTATION];
+export const getTemplateWorkload = (template: V1Template): string =>
```

Comment:
```suggestion
export const getTemplateWorkload = (template: V1Template): WORKLOADS =>
```
you should not change the signature of a common lib method
( if you do, you need to go over the code and make sure it's ok )

Reply:
I changed that because it didn't work, compilation was failing which is obvious if you think deeper about the change we did here. Tried many things and wasn't able to use the better type. But I haven't tried the force cast yet...